src/app/auctioneer-dialog/auctioneer-add-item-dialog.component.ts

```diff
@@ -11,16 +13,30 @@ export class AuctioneerAddItemDialogComponent {
   tagsList: string[] = ['Automotive', 'Books', 'Clothing', 'Electronics', 'Jewelry', 'Kitchen', 'Movies', 'Music', 'Sports'];
 
   missingFields = false;
+  imagePointer: File = null;
+  clientID: string = 'dd65a136a77f9f3';
+  uploadUrl: string = 'https://api.imgur.com/3/image';
 
   constructor(
-    public dialogRef: MatDialogRef<AuctioneerAddItemDialogComponent>,
+    public dialogRef: MatDialogRef<AuctioneerAddItemDialogComponent>, private http:HttpClient,
     @Inject(MAT_DIALOG_DATA) public data: AuctionItem) {
     }
 
   onNoClick(): void {
     this.dialogRef.close();
   }
 
+  async onFileSelected(event) {
```

Comment:
This could lead to a race condition. If the user creates the item before a response is returned from imgur, the url wouldnt be saved. Upload to imgur should happen after the user clicks 'Add Item' and the modal should wait for a response before it updates 'data' and saves to mongo